src/effects/VST/VSTEffect.cpp

```diff
@@ -3415,15 +3548,22 @@ void VSTEffectWrapper::ForEachParameter(ParameterVisitor visitor) const
 }
 
 
-bool VSTEffectWrapper::FetchSettings(VSTEffectSettings& vstSettings) const
+bool VSTEffectWrapper::FetchSettings(VSTEffectSettings& vstSettings, bool doFetch) const
 {
    // Get the fallback ID-value parameters
    ForEachParameter
    (
       [&](const ParameterInfo& pi)
       {
-         float val = callGetParameter(pi.mID);
-         vstSettings.mParamsMap[pi.mName] = val;
+         if (doFetch)
+         {
+            float val = callGetParameter(pi.mID);
+            vstSettings.mParamsMap[pi.mName] = val;
+         }
+         else
+         {
+            vstSettings.mParamsMap[pi.mName] = std::nullopt;
+         }
```

Comment:
This too looks all right.

However there is no remembering of the mID value from ParameterInfo.

I think the Message may need to have that when the values are ultimately used in RealtimeProcessStart, so you can do callSetParameter.

So maybe it will be easiest to change the data type of the map to something like `std::optional<std::pair<int, double>>`